android/src/main/java/com/swmansion/gesturehandler/core/LongPressGestureHandler.kt

```diff
@@ -46,36 +91,72 @@ class LongPressGestureHandler(context: Context) : GestureHandler<LongPressGestur
       previousTime = SystemClock.uptimeMillis()
       startTime = previousTime
       begin()
-      startX = sourceEvent.rawX
-      startY = sourceEvent.rawY
+      startX = getAverageX(sourceEvent)
+      startY = getAverageY(sourceEvent)
+
+      ++currentPointers
+    }
+
+    if (sourceEvent.actionMasked == MotionEvent.ACTION_POINTER_DOWN) {
+      startX = getAverageX(sourceEvent)
+      startY = getAverageY(sourceEvent)
+
+      if (++currentPointers > numberOfPointersRequired) {
+        fail()
+        currentPointers = 0
+      }
+    }
+    if (state == STATE_BEGAN && currentPointers == numberOfPointersRequired && !hasScheduledActivation) {
       handler = Handler(Looper.getMainLooper())
       if (minDurationMs > 0) {
         handler!!.postDelayed({ activate() }, minDurationMs)
       } else if (minDurationMs == 0L) {
         activate()
       }
+
+      hasScheduledActivation = true
     }
     if (sourceEvent.actionMasked == MotionEvent.ACTION_UP || sourceEvent.actionMasked == MotionEvent.ACTION_BUTTON_RELEASE) {
+      --currentPointers
+
       handler?.let {
         it.removeCallbacksAndMessages(null)
         handler = null
       }
+
       if (state == STATE_ACTIVE) {
         end()
       } else {
         fail()
       }
+
+      hasScheduledActivation = false
+    } else if (sourceEvent.actionMasked == MotionEvent.ACTION_POINTER_UP) {
+      if (--currentPointers < numberOfPointersRequired && state != STATE_ACTIVE) {
+        fail()
+        currentPointers = 0
+        hasScheduledActivation = false
+      } else {
+        startX = getAverageX(sourceEvent, true)
+        startY = getAverageY(sourceEvent, true)
```

Comment:
I'm not sure if we want to change `startX` and `startY` in this case, but if we don't, releasing a pointer will basically cancel handler (since it is very hard to avoid little 1px movements) (cc @j-piasecki)

Reply:
I don't think I see what you mean.

> releasing a pointer will basically cancel handler

Isn't that what we want? If there are more pointers than set by the user we fail the handler so at this point there should always be too few pointers, no?

Reply:
It depends what do we understand by `numberOfPointers` are those pointers required for activation, or through whole state changes process? 

Also, the fact that I've introduced this change is based on this:

>it is very hard to avoid little 1px movements

What it really means is that if you lift both fingers at once, you'll most likely receive `ACTION_POINTER_UP` followed by `ACTION_MOVE`. Now, if we don't change starting point it will definitely exceed `maxDistance`, which means that the handler will fail even though we had enough pointers.

Reply:
> Isn't that the same thing? I thought that it would require the exact amount of pointers set to activate and it would fail when adding or removing them before activation (so no activate when too few and no activate when too many).

By "whole state changes process" I meant from `UNDETERMINED` to one of the final states. I'm not talking only about activation 😅 The main reason why I did this is because it is almost impossible to end gesture if those values are not updated - it will fail because `maxDistance` will be exceeded (especially if the pointers are far away from each other).